crates/gitbutler-core/src/virtual_branches/virtual.rs

```diff
@@ -1916,11 +1941,16 @@ fn get_applied_status(
         .position(|b| b.selected_for_changes == Some(max_selected_for_changes))
         .unwrap_or(0);
 
+    // Everything claimed has been removed from `base_diffs`, here we just
+    // process the remaining ones.
     for (filepath, hunks) in base_diffs {
         for hunk in hunks {
             let hash = Hunk::hash_diff(hunk.diff_lines.as_ref());
-            let vbranch_pos = if let Some(locked_to) = git_hunk_map.get(&hash) {
-                let p = virtual_branches.iter().position(|vb| vb.id == *locked_to);
+            let locked_to = locked_hunk_map.get(&hash);
+
+            let vbranch_pos = if let Some(locked_to) = locked_to {
+                let branch_id = commit_to_branch.get(&locked_to[0]).unwrap();
```

Comment:
This should never happen, but we should probably fail gracefully and warn that it happened.